### ISSUE TITLE
Initial setup is in Dutch

### ISSUE THREAD

**Issue by thany** [zenphoto/zenphoto#1215]
Ok, I can read Dutch. That's not the issue. The issue is that nothing should tell zenphoto to pick the Dutch language. There's nothing in my OS, browser, or otherwise, that could possibly dictate websites to use the Dutch language. *Locale* maybe, but not language.

My OS language is English/UK, my browser language is English/UK, and my locale is Dutch/NL.
So, number and currency and datetime formatting is Dutch, and words are English.

TL;DR zenphoto incorrectly assumes locale==language.

After learning this lesson, will you please fix this?
What if someone accesses their gallery at a public internet pc in a hotel in Japan (or another country where you definitely can't read the local language)? Even if that pc is set to English *language*, it'll certainly be set to Japanese *locale*. Good luck figuring out how to get around then.

**Comment by thany** [comment]
Also, the US flag is selected at the bottom. So it even got that wrong...

**Comment by thany** [comment]
Even when I set the user to US English, AND I set the language in the options to US English, the whole admin site is still in Dutch...

**Comment by thany** [comment]
Okay, it's getting weird now. Even when I delete the *whole* `zp-core/locale` directory, the site is still in Dutch. I guess Dutch is the default or something?
But if I leave `en_US` in there, it still insists on displaying Dutch.

Ik had niet gedacht dat zenphoto zoveel van mijn taal hield 😘

**Comment by acrylian** [comment]
Check the locales your sever has. If you are using MAMP you should be covered

> TL;DR zenphoto incorrectly assumes locale==language.

See the other ticket. See #1207

**Comment by kochs-online** [comment]
> I guess Dutch is the default or something?

Maybe Dutch is chosen because it's the first language in the (alphabetically ordered) list?

**Comment by acrylian** [comment]
I have no idea right now. Default language without gettext and such is of coure en_US.

**Comment by thany** [comment]
Zenphoto mustn't care about what locales the server supports. Locale is not language.

**Comment by acrylian** [comment]
> Zenphoto mustn't care about what locales the server supports. Locale is not language.

Gettext does.

**Comment by kochs-online** [comment]
I have to agree with acrylian on this. Locale is **required** by PHP for `gettext()` to translate texts (cp. [PHP documentation examples](https://php.net/manual/en/function.gettext.php#refsect1-function.gettext-examples)). So you **need** locales to support language translations.

Also you must not mix up different terms in this discussion. There is "locale" in a geographic sense like "region" that of course does not define the language someone inside this region uses. And then there's "locale" in computing and – [to quote Wikipedia](https://en.wikipedia.org/wiki/Locale_(computer_software)) – "[i]n computing, a _locale_ is a set of parameters that defines the **user's language, region and any special variant preferences** that the user wants to see in their user interface".

**Comment by thany** [comment]
Language is only part of the locale if the locale is specified as such. I don't have the time to dive into the spec right now, but after "xx-XX" can come a couple of subtags that define other settings to use along with the locale.

If the locale is only "en-US" it is ambiguous of language and must not make any assumptions. In the same way that "zh-CN" (Chinese culture in China is ambiguous of language, too - it could be Cantonese, Mandarin, English, or even Portuguese. Or some of the other official languages. Just like in the "mt-MT" is the Maltese locale in Malta, and the language could be either English or Maltese.

Language CAN be specified in the locale, but not if that locale only consists of a culture & country, which in zenphoto it does.

In the same way, locale should not be tied to a currency either. Nor to a specific geographical location. Nor to a single timezone.

**Comment by acrylian** [comment]
> which in zenphoto it does.

This is how PHP gettext works if you like or not. If Zenphoto would ignore this you still would not get any translation  without the right locale installed. We are not talking about geopgraphical terms here. Then we also would need to talk about using flags in the selector which is also not really correct. Still anyone gets what is meant.

Please let's get back to your actual problem.
